src/App.js

```diff
@@ -103,7 +102,7 @@ class Root extends React.Component {
   }
 
   handleWebsocket = (wsData) => {
-    if (hathorLib.wallet.loaded()) {
+    if (hathorLib.wallet.loaded() && !this.props.loadingAddresses) {
```

Comment:
What changed in the lib that we need this new condition?

Reply:
This was already a bug, in my opinion.

Just like the if added in the `reloadData` websocket message, it makes no sense to handle a ws message (like a new tx message) if it's still loading data.

Reply:
I thought `hathorLib.wallet.loaded()` was checking exactly that, but it only checks if the wallets was initialized, right? It is not related to the connection and its state.

Reply:
I think we should handle messages while we are loading the addresses, otherwise a race condition may occur. For example: Let's say the wallet has only one address, and we just requested it to the server.

1) Wallet requests the transaction list
2) The server receives the request and sends the response
3) A new transaction arrives in the server, and a message is sent in the websocket.
4) The websocket message arrives in the wallet.
5) The response arrives in the wallet.

Notice that, even thought the response was sent before, it arrived later. We cannot guarantee it would arrive first because the HTTP and the Websocket are two different connections. In this case, if we discard the websocket message, we would show the wrong balance to the user.

Reply:
Yes, the `hathorLib.wallet.loaded()` checks only if the wallet has an `xpub`, i.e. if it was already created.

I agree with you, it's important to handle new messages while loading the addresses.

Removed that condition from the `if`.

Reply:
Ok. Thanks!